submissions/balancer-design/universal_8to8.md

```diff
@@ -0,0 +1,88 @@
+
+![Auto-looping 4-4](autoloops_compressed.jpg)
+
+After some testing, we discover a major issue. If the looping items try to enter a belt that's already full, they just back up all the way to the split-off point and block the overflow, putting us back to square one:
+
+![Broken auto-looping 4-4](failed_autoloops_compressed.jpg)
+
+To remedy this, we need to make sure every item on the loop can get to every single one of the loop's exits. This can be done in many ways, but for our needs we're looking for the smallest number of splitters. It just so happens that the most splitter-economical way to do this is with another balancer. Let's add it to the loop:
+
+![Fixed 4-4](secondary_balancer_compressed.jpg)
+
+We now have a fully functional universal 4-4. However, it can be improved. Because of certain balancer mechanics, we don't actually need 4 belts looping back. In general, we need 3 belts less than our balancer is made for. This means we can optimize our 4-4 by compressing the looping belts into 2, and then splitting them again:
+
+![Simpler universal 4-4](simplified_balancer_compressed.jpg)
+
+Now we can shuffle the components around until we get something small, such as this:
+
+![Small universal 4-4](small_balancer_compressed.jpg)
+
+This balancer, while pretty long, is only 6 wide, which means you can fit it in a standard 4 belt bus, and have just enough space to tile them next to each other. It's important to note that this is not my design, although it's based on one of mine. Sadly, I've lost the name of the designer, so I can't give proper credit.
+
+### Advanced design
+
+Now, let's try something bigger: a universal 8-8. We begin by doing the same steps as with our 4-4, except we use 8-8 balancers instead. This is what we get:
+
+![Basic universal 8-8](simple_8-8_compressed.jpg)
+
+Unfortunately, this is an imperfect balancer. Here is one condition which makes it bottleneck:
+
+![Basic 8-8 bottlenecking](failed_8-8_compressed.jpg)
+
+As you can see, 6 belts of input and output are available, but less than 6 are coming through. This is a problem. The standard 8-8 balancer is throughput-limited, meaning the entire thing is as well. Luckily, this is easily fixed by simply replacing the core balancer with a throughput-unlimited one:
+
+![Proper universal 8-8](TU_8-8_compressed.jpg)
+
+Notice how the secondary balancer can stay throughput-limited, because its only job is to allow items access to all belts. Now, we could try to use our loopback compressing trick, but we'd need a throughput unlimited 8-5 merger and a 5-8 splitter. While possible, this will make us use a lot more splitters, making everything more complex. Now let's shuffle the components around, and also use some belt weaving to compress 8 belts into 4 tiles:
```

Comment:
`splitter` is confused with `splitter`. Maybe `belt splitter`? That doesn't work. Maybe you could just write it out.

> but we'd need a throughput unlimited way to merge 8 belts to 5 belts _and_ a way to divide 5 belts over 8 output belts.